### ISSUE TITLE
RCS1235 for changing Add to AddRange does not appear to do type check for the parameter of AddRange

### ISSUE THREAD

**Issue by nvmkpk** [JosefPihrt/Roslynator#894]
**Product and Version Used**: 
Rolsynator 2022 extension version 4.1.0

**Steps to Reproduce**:
Create a console project with a code file containing below code:
```csharp
public void Main()
{
    var collection = new ObjectCollection();
    var dictionary = new Dictionary<string, object>();

    foreach (var item in dictionary.Values)
    {
        collection.Add(item);  // RCS1235 is reported here
    }
}

class ObjectCollection : Collection<object>
{
    public void AddRange(ObjectCollection other)
    {
    }
}
```

**Actual Behavior**:
RCS1235 is suggested. If you apply the code fix, the code will fail to compile with below error:
```
CS1503	Argument 1: cannot convert from 'System.Collections.Generic.Dictionary<string, System.CodeDom.CodeNamespace>.ValueCollection' to 'ObjectCollection'
```

**Expected Behavior**:
RCS1235 should not be suggested.
